int16[] indices
string label